### PR TITLE
Fix compilations errors

### PR DESCRIPTION
**Pull request by anwar6953**
Implement newPluginInstance(String,MacroEvaluator) in MockRealtimeContext and MockTransformContext, to match new APIs in cdap.

`mvn clean package -DskipTests` passed locally.

### CODE REVIEW

File: hydrator-common/src/main/java/co/cask/hydrator/common/test/MockRealtimeContext.java [hunk 1]
```diff
@@ -79,6 +81,12 @@ public PluginProperties getPluginProperties(String pluginId) {
   }
 
   @Override
+  public <T> T newPluginInstance(String s,
+                                 MacroEvaluator macroEvaluator) throws InstantiationException, InvalidMacroException {
+    return null;
```
Comment (Nishith) [inline hydrator-common/src/main/java/co/cask/hydrator/common/test/MockRealtimeContext.java:86]: Move "MacroEvaluator macroEvaluator)" to the previous line?

File: hydrator-common/src/main/java/co/cask/hydrator/common/test/MockTransformContext.java [hunk 2]
```diff
@@ -83,6 +85,12 @@ public String getStageName() {
   }
 
   @Override
+  public <T> T newPluginInstance(String s,
+                                 MacroEvaluator macroEvaluator) throws InstantiationException, InvalidMacroException {
```
Comment (Nishith) [inline hydrator-common/src/main/java/co/cask/hydrator/common/test/MockTransformContext.java:89]: Move "MacroEvaluator macroEvaluator)" to the previous line?

**Comment by Nishith** [comment]
LGTM. Wait for the build to pass?

File: hydrator-common/src/main/java/co/cask/hydrator/common/test/MockRealtimeContext.java (hunk 1 above)
Comment (anwar6953) [inline hydrator-common/src/main/java/co/cask/hydrator/common/test/MockRealtimeContext.java:86]: then, I'd have to split the exceptions to the next line. I just thought the split here looked cleaner.

**Comment by anwar6953** [comment]
This is just fixing compilation issues, so just made sure that it compiles now.
Merging to unblock next run of integration tests.

**Pull request merged**